tests/test_config.py

```diff
@@ -121,3 +123,61 @@ def test_basics(path, glue_database, glue_table, workgroup0, workgroup1):
 def test_athena_cache_configuration():
     wr.config.max_local_cache_entries = 20
     assert wr.config.max_remote_cache_entries == 20
+
+
+def test_botocore_config(path):
+    original = botocore.client.ClientCreator.create_client
+
+    # Default values for botocore.config.Config
+    expected_max_retries_attempt = 5
+    expected_connect_timeout = 10
+    expected_max_pool_connections = 10
+    expected_retry_mode = ""
```

Comment:
default is `retry_mode="legacy"` if you want to be explicit alongside `max_attempts=5`

Reply:
The problem is that `retry_mode` is just not set in the config if it isn't explicitly defined and I wouldn't set it explicitly because of the reasoning here: https://github.com/awslabs/aws-data-wrangler/pull/535#discussion_r567468983.